### DISCUSSION TITLE
Indicate commit order / direction in PR "commits" tab

### DISCUSSION THREAD

**Discussion by stevenroose** [community/community#20863] [Pull Requests]
I always struggle to know in what order to read commits in the "commits" tab. `git` itself orders older ones below, but various other platforms have different orderings. I don't think it makes sense to push for a certain order, but the order chosen should at least be clearly indicated. (I would expect without indication the default would be to mimick `git`, but GitHub seems to do the opposite without indicating so.) An important detail here is that dates aren't a solid indicator as dates can be anything, it's the order in which the commits are chained that should be indicated. Also, often multiple commits share the same date anyway.

Often, the CI run icon can indicate which commit is the latest and thus the order, but that's an unreliable method.

This is how it currently looks:

![image](https://user-images.githubusercontent.com/853468/179729899-cd653c33-ddb0-4eb5-841e-08d01fbdb654.png)

The line going through the commit clearly seems to suggest that on both sides the chain continues. Just removing the extra line at the "end" would already be a good indicator. Even better would be to make it a dotted line on the "start" side, i.e. where it goes back to the merge-base with master. Better would be a little more explicit, like emphasizing the "tip"/"end" with a bolder dot or something.

But in the current view, it's impossible to know in which order the commits are to be applied other than checking them out in `git`. (Even worse, when you click on one of them, no "parent" is indicated. Nor do the "previous" and "next" buttons give any indicator because they don't go back outside the scope of the PR.)

**Comment by maximetinu** [comment]
I also struggle with this a lot bc GitHub orders differently the commits of a repo page (top is newest, bottom is oldest) compared to the commits of a PR page (top is oldest, bottom is newest)

Making it configurable is the ideal solution, but at least we need consistency! 😂

**Comment by xiaoboa** [comment]
made a tamper monkey script for workaround
https://gist.github.com/xiaoboa/703e1fa3fc9b99692c1a1ff2f0b42ccd
